python/ray/air/result.py

```diff
@@ -186,16 +187,10 @@ def from_path(cls, path: Union[str, os.PathLike]) -> "Result":
         checkpoint_dirs = sorted(local_path.glob("checkpoint_*"))
 
         if checkpoint_dirs:
-            if _use_storage_context():
-                checkpoints = [
-                    NewCheckpoint.from_directory(checkpoint_dir)
-                    for checkpoint_dir in checkpoint_dirs
-                ]
-            else:
-                checkpoints = [
-                    Checkpoint.from_directory(checkpoint_dir)
-                    for checkpoint_dir in checkpoint_dirs
-                ]
```

Comment:
could we actually keep all the branched paths for the moment? I want to save all code deletion for after the entire migration is complete.

Reply:
So, this PR should only touch examples, tests, and places where the usage is identical (aka `Checkpoint.from_directory`)

Reply:
Oh yeah sure, I was also questioning whether or not to include those in this PR.

But just to double check - this particular piece of code in `Result` is implementation but the usage is indentical, would you want me to unify this in this PR or not?

Reply:
oh yes this one is fine